pkg/filesystem/path/unix_parser.go

```diff
@@ -49,33 +49,34 @@ func (p unixParser) ParseScope(scopeWalker ScopeWalker) (next ComponentWalker, r
 			return nil, nil, err
 		}
 
-		return next, unixRelativeParser{stripOneOrMoreSlashes(p.path)}, nil
+		return next, relativeParser{stripOneOrMoreSlashes(p.path), '/'}, nil
 	}
 
 	next, err = scopeWalker.OnRelative()
 	if err != nil {
 		return nil, nil, err
 	}
 
-	return next, unixRelativeParser{p.path}, nil
+	return next, relativeParser{p.path, '/'}, nil
 }
 
-type unixRelativeParser struct {
-	path string
+type relativeParser struct {
+	path          string
+	pathSeparator byte
```

Comment:
I thought that Windows always accepted paths that either contain backslashes or forward slashes, or a mixture thereof. My recommendation would be to give the Windows parser its own `relativeParser` that checks both.

Reply:
Aha, I did not think of a mixture of the two. Do we want to fully support that now? With a new relative parser we improve the state of affairs, but I think more changes are required for it to work entirely.